cmd/live_udp_reader/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/eluv-io/avpipe/broadcastproto/transport"
+	"golang.org/x/net/ipv4"
+)
+
+func main() {
+	url := flag.String("url", "", "UDP URL to read (e.g. udp://232.1.2.3:1234)")
+	bufSize := flag.Int("buf", transport.PacketSize*7, "read buffer size in bytes")
+	flag.Parse()
+
+	if *url == "" {
+		fmt.Println("Usage: live_udp_reader -url udp://232.1.2.3:1234")
+		os.Exit(1)
+	}
+
+	log.Printf("Starting live UDP reader for %s", *url)
+
+	udp := transport.NewUDPTransport(*url)
+	rc, err := udp.Open()
+	if err != nil {
+		log.Fatalf("failed to open UDP transport: %v", err)
+	}
+	defer rc.Close()
+
+	buf := make([]byte, *bufSize)
+	oobBuf := make([]byte, 64)
```

Comment:
The 64-byte oobBuf should be sufficient for control messages, but consider using ipv4.ControlMessageSpace for the exact size needed?